src/services/document-service.js

```diff
@@ -186,6 +184,38 @@ class DocumentService {
     }
   }
 
+  async restoreDocumentRevision({ documentKey, revisionId, user }) {
+    if (!user || !user._id) {
+      throw new Error('No user specified');
+    }
+
+    const revisionToRestore = await this.getDocumentRevisionById(revisionId);
+    if (revisionToRestore?.key !== documentKey) {
+      throw new Error(`Revision ${revisionId} is not valid`);
+    }
+
+    const latestRevision = await this.getCurrentDocumentRevisionByKey(documentKey);
+    if (revisionToRestore._id === latestRevision._id) {
+      throw new Error(`Revision ${revisionId} cannot be restored, it is the latest revision`);
```

Comment:
Or we could also just

`return this.getAllDocumentRevisionsByKey(documentKey);` ?

instead of throwing 🤷‍♀️  Just an idea

Reply:
I guess it depends on how we treat it in the frontend. But makes sense, in the end the client/frontend should decide what to do here.
So maybe when you want to restore a revision, you'll simply can't select the last one?

Reply:
P.S. if you agree to do smth like this, we could also have a follow-up ticket for it and get this merged today.